xcube/webapi/context.py

```diff
@@ -269,8 +273,110 @@ def get_dataset_configs(self) -> List[DatasetConfigDict]:
                 dataset_configs += \
                     self.get_dataset_configs_from_stores()
                 self._dataset_configs = dataset_configs
+                self._maybe_assign_store_instance_ids()
         return self._dataset_configs
 
+    def _maybe_assign_store_instance_ids(self):
```

Comment:
Long, complex method. Please refactor and/or add comments so one can understand the algorithm implementation. It takes me too much time to reverse-engineer the intentions coded here.

Reply:
Okay, I tried.